src/components/HelFormFields/HelKeywordSelector/HelKeywordSelector.js

```diff
@@ -0,0 +1,110 @@
+import {HelLabeledCheckboxGroup, HelSelect} from '../index'
+import {FormattedMessage} from 'react-intl'
+import SelectedKeywords from '../../SelectedKeywords/SelectedKeywords'
+import React from 'react'
+import {SideField} from '../../FormFields'
+import {get, uniqBy} from 'lodash'
+import {mapKeywordSetToForm} from '../../../utils/apiDataMapping'
+import {setData as setDataAction} from '../../../actions/editor'
+import PropTypes from 'prop-types'
+import {connect} from 'react-redux'
+import {CopyToClipboard} from 'react-copy-to-clipboard'
+import {ContentCopy} from 'material-ui-icons'
+
+const HelKeywordSelector = (props) => {
+    const {intl, setDirtyState, setData} = props
+    const {values, keywordSets, validationErrors} = props.editor
+
+    const keywords = get(values, 'keywords', [])
+    const mainCategoryOptions = mapKeywordSetToForm(keywordSets, 'helsinki:topics')
+
+    const handleKeywordChange = (checkedOptions) => {
```

Comment:
I would try to move these functions outside the parent function just to keep them pure and not depend on the parent function's scope. Pure functions are safer and cause fewer unwanted side effects since their dependencies are explicitly defined as arguments and not implicitly referenced from the surrounding scope.

Reply:
👍